File: 2023/tests/day15.test.js
const fs = require('fs');
const { part1, part2 } = require('../src/day15');

const example1 = `
HASH
`.trim();
const example2 = `
rn=1,cm-,qp=3,cm=2,qp-,pc=4,ot=9,ab=5,pc-,pc=6,ot=7
`.trim();

const input = fs.readFileSync('inputs/input15.txt', 'utf8');

describe('part 1', () => {
  test('example 1', () => {
    expect(part1(example1)).toBe(52);
  });

  test('example 2', () => {
    expect(part1(example2)).toBe(1320);
  });

  test('solution', () => {
    expect(part1(input)).toBe(513214);
  });
});

describe('part 2', () => {
  test('example 2', () => {
    expect(part2(example2)).toBe(145);
  });

  test('solution', () => {
    expect(part2(input)).toBe(258826);
  });
});

File: 2024/AdventOfCode/Tests/Day15Test.cs
using NUnit.Framework;

namespace AdventOfCode.Tests;

[TestFixture]
public class Day15Test {
    private const string Example = "Inputs/15-Example.txt";
    private const string Example2 = "Inputs/15-Example2.txt";
    private const string Example3 = "Inputs/15-Example3.txt";
    private const string Input = "Inputs/15.txt";
    private static Day15 _day;

    [TestFixture]
    public class Part1 {
        [Test]
        public void TestExample() {
            _day = new Day15(Example);
            const string expected = "10092";
            var solution = _day.Solve_1();
            Assert.That(solution.Result, Is.EqualTo(expected));
        }

        [Test]
        public void TestExample2() {
            _day = new Day15(Example2);
            const string expected = "2028";
            var solution = _day.Solve_1();
            Assert.That(solution.Result, Is.EqualTo(expected));
        }

        [Test]
        public void TestInput() {
            _day = new Day15(Input);
            const string expected = "1478649";
            var solution = _day.Solve_1();
            Assert.That(solution.Result, Is.EqualTo(expected));
        }
    }

    [TestFixture]
    public class Part2 {
        [Test]
        public void TestExample() {
            _day = new Day15(Example);
            const string expected = "9021";
            var solution = _day.Solve_2();
            Assert.That(solution.Result, Is.EqualTo(expected));
        }

        [Test]
        public void TestExample2() {
            _day = new Day15(Example3);
            const string expected = "618";
            var solution = _day.Solve_2();
            Assert.That(solution.Result, Is.EqualTo(expected));
        }

        [Test]
        public void TestInput() {
            _day = new Day15(Input);
            const string expected = "1495455";
            var solution = _day.Solve_2();
            Assert.That(solution.Result, Is.EqualTo(expected));
        }
    }
}